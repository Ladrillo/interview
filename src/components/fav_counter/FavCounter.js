import React, { Component } from 'react';
import styled from 'styled-components';


export default function FavCounter({ count, clickHandler }) {
    return (
        <Styled className='fav-counter' onClick={clickHandler}>
            <span className='fav-counter-heart'>&#9825;</span>
            <span>{count}</span>
        </Styled>
    );
}

const Styled = styled.div`
    width: 100%;
    height: 40px;
    display: flex;
    justify-content: center;
    position: relative;
    margin-top: 40px;
    cursor: pointer;

    .fav-counter-heart {
        font-size: 52px;
        color: #FDF253;
        position: absolute;
        top: -30px;
    }
`;