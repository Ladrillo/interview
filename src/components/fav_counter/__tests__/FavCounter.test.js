import FavCounter from '../FavCounter';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';


describe('FavCounter', () => {

    it('renders correctly', () => {
        const rendered = renderer.create(
            <FavCounter
                count        = {32}
                clickHandler = {() => ({})}
            />
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });
});
