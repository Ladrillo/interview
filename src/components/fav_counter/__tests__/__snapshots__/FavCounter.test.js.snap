exports[`FavCounter renders correctly 1`] = `
<div
  className="fav-counter sc-bdVaJa bLjmXh"
  onClick={[Function]}>
  <span
    className="fav-counter-heart">
    â™¡
  </span>
  <span>
    32
  </span>
</div>
`;
